Authentication failure messages in logs with auth-enabled Standalone mode

**Problem description**
1. Enable ``auth`` in ``standalone-config.properties``:
```
singlenode.enableAuth=true
singlenode.userName=admin
singlenode.passwd=1111_aaaa
singlenode.passwdFile=../config/passwd
```
2. Start Pravega in standalone mode: ``./gradlew startStandalone``

3. Observe stack traces like the following showing up again and again in the logs: 
```
[INFO ] 2019-01-18 00:00:09.964 [core-17] i.p.s.s.h.ZKSegmentContainerMonitor - Container Changes: Desired = [0, 1, 2, 3], Current = [0, 1, 2, 3], PendingTasks = [], ToStart = [], ToStop = [].
[INFO ] 2019-01-18 00:00:10.274 [grpc-default-executor-12] i.p.c.s.r.g.v.ControllerServiceImpl - getCurrentSegments called for stream _system/_requeststream.
[ERROR] 2019-01-18 00:00:10.275 [grpc-default-executor-12] i.p.c.s.r.g.v.ControllerServiceImpl - Controller api failed with authenticator error
[WARN ] 2019-01-18 00:00:10.278 [grpc-default-executor-13] i.p.c.s.i.ControllerImpl - gRPC call for getCurrentSegments with trace id 0 failed with server error.
io.grpc.StatusRuntimeException: UNAUTHENTICATED: Authentication failed
	at io.grpc.Status.asRuntimeException(Status.java:526)
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:418)
	at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:41)
	at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:41)
	at io.grpc.internal.CensusStatsModule$StatsClientInterceptor$1$1.onClose(CensusStatsModule.java:663)
	at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:41)
	at io.grpc.internal.CensusTracingModule$TracingClientInterceptor$1$1.onClose(CensusTracingModule.java:392)
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:443)
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:63)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.close(ClientCallImpl.java:525)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.access$600(ClientCallImpl.java:446)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:557)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[INFO ] 2019-01-18 00:00:10.309 [core-9] i.p.s.s.c.StreamSegmentContainerMetadata - SegmentContainer[1]: EvictedStreamSegments Count = 0, Active = 3
[INFO ] 2019-01-18 00:00:10.309 [core-10] i.p.s.s.c.StreamSegmentContainerMetadata - SegmentContainer[0]: EvictedStre
```

Note: 
- Even with these exceptions, reader/writer client applications work properly; This is true with Pravega client built using the latest code. I got similar exception messages on the client side too with Pravega client of version 0.3.2, and with that version, the client-side operations also failed. 
- @adrianmo has mentioned the same issue in PR #3235.  

**Problem location**
``ControllerServiceImpl.authenticateExecuteAndProcessResults(..)``

**Suggestions for an improvement**

--- a/standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java
+++ b/standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java
@@ -10,6 +10,8 @@
 package io.pravega.local;
 
 import com.google.common.base.Preconditions;
+import io.pravega.client.stream.impl.Credentials;
+import io.pravega.client.stream.impl.DefaultCredentials;
 import io.pravega.common.auth.ZKTLSUtils;
 import com.google.common.base.Strings;
 import io.pravega.controller.server.ControllerServiceConfig;
@@ -40,7 +42,6 @@ import java.io.IOException;
 import java.net.URI;
 import java.util.Arrays;
 import java.util.Optional;
-import java.util.Properties;
 import java.util.UUID;
 import javax.annotation.concurrent.GuardedBy;
 import lombok.Builder;
@@ -256,15 +257,13 @@ public class InProcPravegaCluster implements AutoCloseable {
      * @param segmentStoreId id of the SegmentStore.
      */
     private void startLocalSegmentStore(int segmentStoreId) throws Exception {
-        Properties authProps = new Properties();
-        authProps.setProperty("pravega.client.auth.method", "Default");
-        authProps.setProperty("pravega.client.auth.userName", "arvind");
-        authProps.setProperty("pravega.client.auth.password", "1111_aaaa");
+        if (this.enableAuth) {
+            setAuthSystemProperties();
+        }
 
         ServiceBuilderConfig.Builder configBuilder = ServiceBuilderConfig
                 .builder()
                 .include(System.getProperties())
-                .include(authProps)
                 .include(ServiceConfig.builder()
                         .with(ServiceConfig.CONTAINER_COUNT, containerCount)
                         .with(ServiceConfig.THREAD_POOL_SIZE, THREADPOOL_SIZE)
@@ -302,6 +301,27 @@ public class InProcPravegaCluster implements AutoCloseable {
 
         nodeServiceStarter[segmentStoreId] = new ServiceStarter(configBuilder.build());
         nodeServiceStarter[segmentStoreId].start();
+    }
+
+    private void setAuthSystemProperties() {
+        if (authPropertiesAlreadySet()) {
+            log.debug("Auth params already specified via system properties or environment variables.");
+        } else {
+            if (!Strings.isNullOrEmpty(this.userName)) {
+                Credentials credentials = new DefaultCredentials(this.passwd, this.userName);
+                System.setProperty("pravega.client.auth.loadDynamic", "false");
+                System.setProperty("pravega.client.auth.method", credentials.getAuthenticationType());
+                System.setProperty("pravega.client.auth.token", credentials.getAuthenticationToken());
+                log.debug("Done setting auth params via system properties.");
+            } else {
+                log.debug("Cannot set auth params as username is null or empty");
+            }
+        }
+    }
+
+    private boolean authPropertiesAlreadySet() {
+        return !Strings.isNullOrEmpty(System.getProperty("pravega.client.auth.method"))
+                || !Strings.isNullOrEmpty(System.getenv("pravega_client_auth_method"));
     }
 
     private void startLocalControllers() {

